fog/overseer/server/src/worker.rs

```diff
@@ -0,0 +1,546 @@
+                Err(_) => {
+                    let number_of_remaining_tries = Self::NUMBER_OF_TRIES - current_try as u8;
+                    let error_message = format!("The following key  was not successfully reported as lost: {}. Will try {} more times", inactive_outstanding_key, number_of_remaining_tries);
+                    OperationResult::Retry(OverseerError::ReportLostKey(error_message))
+                }
+            }
+        });
+
+        match result {
+            Ok(success_message) => {
+                log::info!(self.logger, "{}", success_message);
+                Ok(())
+            }
+            // TODO: Add alerting that says we need to manually report these keys
+            // as lost.
+            Err(err) => Err(err.into()),
+        }
+    }
+
+    fn set_new_key_on_a_node(&self) -> Result<usize, OverseerError> {
+        for (i, ingest_client) in self.ingest_clients.iter().enumerate() {
+            let result = retry_with_index(Fixed::from_millis(200), |current_try| {
+                if current_try as u8 > Self::NUMBER_OF_TRIES {
+                    let error_message = format!(
+                        "Did not succeed in setting a new key on node at index {}",
+                        i
+                    );
+                    return OperationResult::Err(OverseerError::SetNewKey(error_message));
+                }
+                match ingest_client.new_keys() {
+                    Ok(_) => {
+                        log::info!(
+                            self.logger,
+                            "New keys successfully set on the ingest node at index {}.",
+                            i
+                        );
+                        OperationResult::Ok(i)
+                    }
+                    // TODO: We'll need to alert Ops to take manual action at this point.
+                    Err(_) => {
+                        let number_of_remaining_tries = Self::NUMBER_OF_TRIES - current_try as u8;
+                        let error_message = format!("New keys were not successfully set on the ingest node at index {}. Will try {} more times.", i, number_of_remaining_tries);
+                        log::warn!(self.logger, "{}", error_message);
+                        OperationResult::Retry(OverseerError::SetNewKey(error_message))
+                    }
+                }
+            });
+
+            match result {
+                Ok(i) => return Ok(i),
+                Err(retry_error) => {
+                    if i == self.ingest_clients.len() {
+                        return Err(retry_error.into());
+                    }
+                }
+            }
+        }
+
+        Err(OverseerError::SetNewKey(
+            "New keys were not successfully set on any of the idle nodes.".to_string(),
+        ))
+    }
+
+    fn activate_a_node(&self, activated_node_index: usize) -> Result<(), OverseerError> {
+        let result = retry_with_index(Fixed::from_millis(200), |current_try| {
```

Comment:
This should probably also only be called on idle nodes.

Reply:
Same as above.